src/lib.rs

```diff
@@ -370,33 +520,101 @@ impl TestHarness {
-        let precommits: Vec<_> = self.network
+        let (block_hash, patch) = {
+            let validator_id = self.leader().validator_id().expect(
+                "Tested node is not a validator",
+            );
+            let transactions = self.mempool();
+            self.blockchain.create_patch(
+                validator_id,
+                height,
+                tx_hashes,
+                &transactions,
+            )
+        };
+
+        // Remove txs from mempool
+        {
+            let mut transactions = self.mempool.write().expect(
+                "Cannot modify transactions in mempool",
+            );
+            for hash in tx_hashes {
+                transactions.remove(hash);
+            }
+        }
+
+        let propose = self.leader().create_propose(height, &last_hash, tx_hashes);
+        let precommits: Vec<_> = self.network()
             .validators()
             .iter()
             .map(|v| v.create_precommit(&propose, &block_hash))
             .collect();
-        handler.commit(block_hash, precommits.iter(), Some(round));
+
+        self.blockchain
+            .commit(
+                &mut self.service_context,
+                &patch,
+                block_hash,
+                precommits.iter(),
+            )
+            .unwrap();
+
+        self.poll_events();
+    }
+
+    /// Update test network configuration if such an update has been scheduled
+    /// with `propose_configuration_change`.
+    fn update_configuration(&mut self) {
+        use ConfigurationProposalState::*;
+
+        let height = self.height();
+        if let Some(cfg_proposal) = self.cfg_proposal.take() {
+            match cfg_proposal {
+                Uncommitted(cfg_proposal) => {
+                    // Commit configuration proposal
+                    let stored = cfg_proposal.stored_configuration().clone();
+                    let mut fork = self.blockchain.fork();
+                    CoreSchema::new(&mut fork).commit_configuration(stored);
+                    let changes = fork.into_patch();
+                    self.blockchain.merge(changes).unwrap();
+                    self.cfg_proposal = Some(ConfigurationProposalState::Committed(cfg_proposal));
+                }
+                Committed(ref cfg_proposal) if cfg_proposal.actual_from() == height => {
+                    // Modify the self configuration
+                    self.network_mut().update(
+                        cfg_proposal.us.clone(),
+                        cfg_proposal.validators.clone(),
+                    );
+                }
+                Committed(cfg_proposal) => {
+                    self.cfg_proposal = Some(ConfigurationProposalState::Committed(cfg_proposal));
+                }
+            }
+        }
+    }
+
+    /// Creates block with the given transactions.
+    /// Transactions that are in mempool will be ignored.
+    pub fn create_block_with_transactions(&mut self, txs: Vec<Box<Transaction>>) {
```

Comment:
Repeating question from #2: what if some of transactions supplied to the method are already committed (i.e., present in one of previous blocks)?

Reply:
Hm, in my opinion that would raises panic.